src/commonMain/kotlin/fr/acinq/lightning/bin/csv/WalletPaymentCsvWriter.kt

```diff
@@ -0,0 +1,126 @@
+                is LightningOutgoingPayment.Details.Normal -> listOf(Details("lightning_sent", amount = -payment.amount, feeCredit = 0.msat, miningFee = 0.sat, serviceFee = payment.fees, paymentHash = payment.paymentHash, txId = null))
+                is LightningOutgoingPayment.Details.SwapOut -> listOf(Details("legacy_swap_out", amount = -payment.amount, feeCredit = 0.msat, miningFee = details.swapOutFee, serviceFee = 0.msat, paymentHash = null, txId = null))
+                is LightningOutgoingPayment.Details.Blinded -> listOf(Details("lightning_sent", amount = -payment.amount, feeCredit = 0.msat, miningFee = 0.sat, serviceFee = payment.fees, paymentHash = payment.paymentHash, txId = null))
+            }
+
+            is SpliceOutgoingPayment -> listOf(Details("splice_out", amount = -payment.amount, feeCredit = 0.msat, miningFee = payment.miningFees, serviceFee = 0.msat, paymentHash = null, txId = payment.txId))
+            is ChannelCloseOutgoingPayment -> listOf(Details("channel_close", amount = -payment.amount, feeCredit = 0.msat, miningFee = payment.miningFees, serviceFee = 0.msat, paymentHash = null, txId = payment.txId))
+            is SpliceCpfpOutgoingPayment -> listOf(Details("fee_bumping", amount = 0.msat, feeCredit = 0.msat, miningFee = payment.miningFees, serviceFee = 0.msat, paymentHash = null, txId = payment.txId))
+            is InboundLiquidityOutgoingPayment -> listOf(
+                Details(
+                    "liquidity",
+                    amount = 0.msat,
+                    feeCredit = -payment.feeCreditUsed,
+                    miningFee = payment.miningFees,
+                    serviceFee = payment.serviceFees.toMilliSatoshi(),
+                    paymentHash = null,
+                    txId = payment.txId
+                )
+            )
+        }
+
+        details.forEach { addRow(timestamp, it) }
+
+    }
+
+    private fun extractLightningPaymentParts(payment: IncomingPayment): List<Details> = payment.received?.receivedWith.orEmpty()
+        .map {
+            when (it) {
+                is IncomingPayment.ReceivedWith.LightningPayment -> Details("lightning_received", amount = it.amountReceived, feeCredit = 0.msat, miningFee = 0.sat, serviceFee = 0.msat, paymentHash = payment.paymentHash, txId = null)
+                is IncomingPayment.ReceivedWith.AddedToFeeCredit -> Details("fee_credit", amount = 0.msat, feeCredit = it.amountReceived, miningFee = 0.sat, serviceFee = 0.msat, paymentHash = payment.paymentHash, txId = null)
+                is IncomingPayment.ReceivedWith.NewChannel -> Details("pay_to_open", amount = it.amountReceived, feeCredit = 0.msat, miningFee = it.miningFee, serviceFee = it.serviceFee, paymentHash = payment.paymentHash, txId = null)
+                is IncomingPayment.ReceivedWith.SpliceIn -> Details("pay_to_splice", amount = it.amountReceived, feeCredit = 0.msat, miningFee = it.miningFee, serviceFee = it.serviceFee, paymentHash = payment.paymentHash, txId = null)
```

Comment:
> Do you really want to keep pay_to_open and pay_to_splice?

Yes, because those are legacy operations and should not be mixed with the new protocol imo.

> Also, we have the txId field in it, is there a reason why you're setting it to null instead of providing its real value?

Good catch!